src/EnergyPlus/Plant/Component.hh

```diff
@@ -162,6 +162,107 @@ namespace DataPlant {
+        DataPlant::CtrlType::Invalid,   //	"ZoneHVAC:LowTemperatureRadiant:VariableFlow"
+        DataPlant::CtrlType::Invalid,   //	"ZoneHVAC:LowTemperatureRadiant:ConstantFlow"
+        DataPlant::CtrlType::Invalid,   //	"AirTerminal:SingleDuct:ConstantVolume:CooledBeam"
+        DataPlant::CtrlType::Invalid,   //	"Coil:Heating:WaterToAirHeatPump:EquationFit"
+        DataPlant::CtrlType::Invalid,   //	"Coil:Cooling:WaterToAirHeatPump:EquationFit"
+        DataPlant::CtrlType::Invalid,   //	"Coil:Heating:WaterToAirHeatPump:ParameterEstimation"
+        DataPlant::CtrlType::Invalid,   //	"Coil:Cooling:WaterToAirHeatPump:ParameterEstimation"
+        DataPlant::CtrlType::HeatingOp, //	"Refrigeration:Condenser:WaterCooled"
+        DataPlant::CtrlType::Invalid,   //	"Refrigeration:CompressorRack"
+        DataPlant::CtrlType::Invalid,   //	"AirLoopHVAC:UnitaryHeatPump:AirToAir:MultiSpeed"
+        DataPlant::CtrlType::DualOp,    //	"ChillerHeater:Absorption:DoubleEffect"
+        DataPlant::CtrlType::Invalid,   //	"PipingSystem:Underground:PipeCircuit"
+        DataPlant::CtrlType::HeatingOp, //	"SolarCollector:IntegralCollectorStorage"
+        DataPlant::CtrlType::Invalid,   //	"Coil:Heating:WaterToAirHeatPump:VariableSpeedEquationFit"
+        DataPlant::CtrlType::Invalid,   //	"Coil:Cooling:WaterToAirHeatPump:VariableSpeedEquationFit"
+        DataPlant::CtrlType::DualOp,    //	"PlantComponent:UserDefined"
+        DataPlant::CtrlType::Invalid,   //	"Coil:UserDefined"
+        DataPlant::CtrlType::Invalid,   //	"ZoneHVAC:ForcedAir:UserDefined"
+        DataPlant::CtrlType::Invalid,   //	"AirTerminal:SingleDuct:UserDefined"
+        DataPlant::CtrlType::Invalid,   //	"AirConditioner:VariableRefrigerantFlow"
+        DataPlant::CtrlType::DualOp,    //	"GroundHeatExchanger:HorizontalTrench"
+        DataPlant::CtrlType::DualOp,    //	"HeatExchanger:FluidToFluid"
+        DataPlant::CtrlType::DualOp,    //	"PlantComponent:TemperatureSource"
+        DataPlant::CtrlType::DualOp,    //	"CentralHeatPumpSystem"
+        DataPlant::CtrlType::Invalid,   //	"AirLoopHVAC:UnitarySystem"
+        DataPlant::CtrlType::HeatingOp, //	"Coil:Cooling:DX:SingleSpeed:ThermalStorage"
+        DataPlant::CtrlType::CoolingOp, //	"CoolingTower:VariableSpeed:Merkel"
+        DataPlant::CtrlType::Invalid,   //	"SwimmingPool:Indoor"
+        DataPlant::CtrlType::DualOp,    //	"GroundHeatExchanger:Slinky"
+        DataPlant::CtrlType::HeatingOp, //	"WaterHeater:HeatPump:WrappedCondenser"
+        DataPlant::CtrlType::Invalid,   //	"AirTerminal:SingleDuct:ConstantVolume:FourPipeBeam"
+        DataPlant::CtrlType::Invalid,   //	"ZoneHVAC:CoolingPanel:RadiantConvective:Water"
+        DataPlant::CtrlType::CoolingOp, //	"HeatPump:PlantLoop:EIR:Cooling"
+        DataPlant::CtrlType::HeatingOp, //	"HeatPump:PlantLoop:EIR:Heating"
+        DataPlant::CtrlType::HeatingOp  //   "DistrictHeating:Steam"
+    };
```

Comment:
I have feelings about another list of things allocated to the same size as the others, but I will stop talking now.